internal/gitutil/gitutil.go

```diff
@@ -32,10 +32,34 @@ import (
 // for remote repos.  Defaults to UserHomeDir/.kpt/repos if unspecified.
 const RepoCacheDirEnv = "KPT_CACHE_DIR"
 
+var DefaultRef = "main"
+
+// SetDefaultRef sets(and returns) the DefaultRef to "master" if master branch exists in
+// remote repository, falls back to "main" if master branch doesn't exist
+func SetDefaultRef(repo string) string {
```

Comment:
I am curious if we should default to `main` first and then `master`. `main` branch seems like the future and will contain the latest changes.

Reply:
Another interesting thing is over time there will be less repos with `master` so maybe when trying to find `main` first we minimize network calls and roundtrips.

Reply:
Currently, the move has just started. So this logic will assume `master` as default. In case `master` doesn't exist we fall back to `main`. In future, we can do the other way round when `main` becomes popular.

Reply:
I think we should default to `master` rather than `main`. There might be existing repos out there that has a branch called `main` from before this change and we don't want to accidentally chose the wrong branch. I think it is less likely that new repos where the default branch is called `main` will end up with random branch called `master`.

Reply:
Sounds good to me.  Defaulting to your best judgement as the core developers and owners of the tool.